static/js/import_wizard2/components/mapping_page.tsx

```diff
@@ -55,9 +56,32 @@ export function MappingPage(props: MappingPageProps): JSX.Element {
   }
 
   useEffect(() => {
-    // TODO(beets): Use server-side detection API.
-    const predictedMapping = getPredictions(props.csvData, placeDetector);
-    setPredictedMapping(predictedMapping);
+    /* Build JSON of column data for detection API Request */
+    const column_ids = new Map(
+      props.csvData.orderedColumns.map((column) => {
+        return [column.columnIdx, column.id];
+      })
+    );
+    const column_headers = new Map(
+      props.csvData.orderedColumns.map((column) => {
+        return [column.columnIdx, column.header];
+      })
+    );
+    const columnParameters = new Map<string, unknown>([
+      ["column_ids", column_ids],
+      ["column_headers", column_headers],
+      ["column_values", props.csvData.columnValuesSampled],
+    ]);
+    /* Use server-side API for column detection */
+    axios
+      .post(`/api/detection/detect_columns`, JSON.stringify(columnParameters))
```

Comment:
have you checked that JSON.stringify(columnParameters) will give you the correct string? I remember running into problems with JSON.stringify on the Map type in the past. If Map type doesn't work, I would suggest just using an object for everything

```
const columnIds = {};
props.csvData.orderedColumns.forEach((column) => {
columnIds[column.columnIdx] = column.id;
});
...
const columnParameters = {
column_ids: columnIds,
...
}
```

Reply:
Replaced Map with Object